Add support for Debian package build

nodiscc opened the pull request:
* Fixes https://github.com/MrS0m30n3/youtube-dl-gui/issues/21
 * Add a desktop launcher (youtube-dl-gui.desktop)
 * Add Debian package build instructions to README, small wording changes (of -> for)
 * Resulting .deb package was successfully installed and tested on Debian 9 Stretch (stable)
 * Building the package requires `debhelper` and `dpkg-dev` packages

-----------------------------

 * The resulting package can be found at https://github.com/nodiscc/dropbin/blob/master/youtube-dl-gui_0.4.0_all.deb - testers welcome
 * The signed `.dsc` file can be found at https://github.com/nodiscc/dropbin/blob/master/youtube-dl-gui_0.4.0.dsc, signed with GPG key `99883828CD9019B8`
 * The Debian package changelog can be updated with `dch -a` (see `man dch`). The email address used in `debian/control` and `debian/changelog` must match an email address for which you have a valid GPG private key to correctly sign the package.

-------------------------------

@MrS0m30n3 you might want to tag the latest commit as `0.4.0` and then create a new [release](https://github.com/MrS0m30n3/youtube-dl-gui/releases), attaching the `.deb` package to the release. 

https://github.com/MrS0m30n3/youtube-dl-gui#downloads could also be updated to point to that package. The package should work with any Debian derivative, including Ubuntu (so you'd no longer need to link to the PPA). Again, testers welcome.

nodiscc commented:
I just ran `lintian` on the Debian package, it reports the following (minor) warnings:

```
W: youtube-dl-gui: description-synopsis-starts-with-article
W: youtube-dl-gui: binary-without-manpage usr/bin/youtube-dl-gui
W: youtube-dl-gui: desktop-entry-invalid-category Multimedia usr/share/applications/youtube-dl-gui.desktop
```

I also noticed the absence of wxpython in the package dependencies. As I already had wxpython installed this caused no problem, but it might - I could not test. I should check the previous PPA package to know how they handle it.

Ping @hotice, I think you are the youtube-dl-gui PPA maintainer, would you be interested in centralizing the packaging effort here? If we could make it work on all Debian derivatives this would be great. I could not test the package on Ubuntu but maybe you'd like to help with that?

MrS0m30n3 commented on debian/control:
```diff
@@ -0,0 +1,16 @@
+Section: web
+Priority: optional
+Maintainer: nodiscc <nodiscc@gmail.com>
+Build-Depends: debhelper (>= 9), python-all
```
You also need to add **dh-python** and **python-wxgtk3.0**.

nodiscc replied:
The proper way to handle this would be building a debian package for twodict. I'll try to do it when I resume work on this. If everything fails, depending on `python-pip` and installing `twodict` in a postinstall script would also work.

MrS0m30n3 commented on debian/rules:
```diff
@@ -0,0 +1,20 @@
+	dh $@ --with python2
+
+override_dh_auto_install:
+	install -Dm 644 $(CURDIR)/youtube_dl_gui/data/icons/hicolor/16x16/apps/youtube-dl-gui.png $(CURDIR)/debian/youtube-dl-gui/usr/share/icons/hicolor/16x16/apps/youtube-dl-gui.png
```
I am sure you dont need the rules for the PNG icons since the *setup.py* can handle those.

MrS0m30n3 commented on debian/control:
```diff
@@ -0,0 +1,16 @@
+
+Package: youtube-dl-gui
+Architecture: all
+Depends: ${shlibs:Depends}, ${misc:Depends}, ${python:Depends}
```
Maybe we should also add the `Recommends: ffmpeg`

nodiscc replied:
Done https://github.com/MrS0m30n3/youtube-dl-gui/pull/189/files#diff-d837a1c17de0268bcea321239ddbed47R13

MrS0m30n3 commented on debian/control:
```diff
@@ -0,0 +1,16 @@
+Section: web
+Priority: optional
+Maintainer: nodiscc <nodiscc@gmail.com>
+Build-Depends: debhelper (>= 9), python-all
```
There is also the **twodict** requirement but there is no debian package for it.
Maybe we should also add **python-pip** since the easiest way to install **twodict** is the PyPi one.

MrS0m30n3 commented:
Just a quick summary:

- [ ] Move **debian/** -> **packaging/**
- [ ] Perform the changes mentioned in the review
- [ ] Add the *.deb under the README.md #downloads section

MrS0m30n3 commented on youtube-dl-gui.desktop:
```diff
@@ -0,0 +1,8 @@
+Name=youtube-dl GUI
+Exec=youtube-dl-gui
+Icon=youtube-dl-gui
+Categories=Network;Multimedia
```
You should also remove the "Mutlimedia" category since `lintian` reports a warning and there is no "Multimedia" category in the FreeDesktop [specification](https://specifications.freedesktop.org/menu-spec/1.0/apa.html).

nodiscc commented:
Thanks for the thorough review :+1: I will take some time to address it in the coming days/weeks

MrS0m30n3 commented:
Yeah, sure take your time.
Also this rule could come in handy when you move the **debian** directory.

```sh
override_dh_auto_install:
    cd ../ && python2 setup.py install --root=$(CURDIR)/debian/youtube-dl-gui/ --no-compile -O0 --force
```

(there are probably better ways to archieve this)

nodiscc commented:
Hi, I addressed most of the review and packaged twodict see https://github.com/MrS0m30n3/twodict/pull/1). About

> Move debian/ -> packaging/

I have proposed a solution in https://github.com/MrS0m30n3/twodict/pull/1 (using a separate branch to store debian packaging work - this is the recommended method), let me know if you agree to use the same method here.

MrS0m30n3 commented:
I have no problem with keeping the **debian** directory in the main dir as long as we use a different branch.

legrostdg commented:
I just noticed this after starting my own package... Which is almost ready now: https://salsa.debian.org/felix-guest/youtube-dl-gui At this point I think I'm going to use my package, it's a shame I did not see yours before...

@nodiscc please use normal procedure when preparing a Debian package: send an ITP.

legrostdg commented:
There remains your .desktop file, so hopefully all is not lost :)

nodiscc commented:
I should know better :) I should just have updated https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=753350 with the URL of this Pull request or https://github.com/MrS0m30n3/youtube-dl-gui/issues/21 and said I was working on it... Nothing is lost it was a good learning opportunity for me, and I'm stuck on https://github.com/MrS0m30n3/twodict/pull/1 with not much time to work on it. So I'm glad you fixed it anyway, thanks!

I'll diff our works to see how you solved the [lintian warnings](https://github.com/MrS0m30n3/twodict/pull/1#issuecomment-343963660) I was getting.
